Physics/Dynamics/ProductionFormFactor.hpp

```diff
@@ -0,0 +1,130 @@
+#include <memory>
+#include <vector>
+
+#include "AbstractDynamicalFunction.hpp"
+#include "Core/Exceptions.hpp"
+#include "Core/Functions.hpp"
+#include "Core/Spin.hpp"
+#include "FormFactor.hpp"
+
+namespace ComPWA {
+namespace Physics {
+namespace Dynamics {
+
+/// The production formfactor is implemented in Blatt-Weisskopf type \f$B_{L}(q)\f$
+/// Ref. Dalitz Plot Analysis, PDG 2012, where 
+/// \f[
+/// \frac{B^{\prime}(q_{0})}{B^{\prime}_{L}(q)} = q^{L}B^{\prime}_{L}(q, q_{0})
+/// \f]
+class ProductionFormFactor
+    : public ComPWA::Physics::Dynamics::AbstractDynamicalFunction {
+
+public:
+  //============ CONSTRUCTION ==================
+  ProductionFormFactor(std::string name,
+                          std::pair<std::string, std::string> daughters,
+                          std::shared_ptr<ComPWA::PartList> partL);
+  virtual ~ProductionFormFactor();
+
+  boost::property_tree::ptree save() const;
+
+  //================ EVALUATION =================
+
+  std::complex<double> evaluate(const ComPWA::DataPoint &point,
+                                unsigned int pos) const;
+
+  /// Dynamical Blatt-Weisskopf function.
+  /// \param mSq Invariant mass squared
+  /// \param ma Mass of daughter particle
+  /// \param mb Mass of daughter particle
+  /// \param L Orbital angular momentum between two daughters a and b
+  /// \param mesonRadius Meson Radius
+  /// \param ffType Form factor type
+  static std::complex<double>
+  dynamicalFunction(double mSq, double ma, double mb, unsigned int L, 
+                    double mesonRadius, FormFactorType ffType);
+
+  //============ SET/GET =================
+
+  void SetOrbitalAngularMomentum(const ComPWA::Spin &L_) { L = L_; }
+
+  void SetMesonRadiusParameter(std::shared_ptr<ComPWA::FitParameter> r) {
+    MesonRadius = r;
+  }
+
+  std::shared_ptr<ComPWA::FitParameter> GetMesonRadiusParameter() {
+    return MesonRadius;
+  }
+
+  /// \see GetMesonRadius() const { return MesonRadius->value(); }
+  void SetMesonRadius(double w) { MesonRadius->setValue(w); }
+
+  /// Get meson radius.
+  /// The meson radius is a measure of the size of the resonant state. It is
+  /// used to calculate the angular momentum barrier factors.
+  double GetMesonRadius() const { return MesonRadius->value(); }
+
+  /// \see GetFormFactorType()
+  void SetFormFactorType(FormFactorType t) { FFType = t; }
+
+  /// Get form factor type.
+  /// The type of formfactor that is used to calculate the angular momentum
+  /// barrier factors.
+  FormFactorType GetFormFactorType() const { return FFType; }
+
+  void updateParametersFrom(const ParameterList &list);
+  void addUniqueParametersTo(ParameterList &list);
+  void addFitParametersTo(std::vector<double> &FitParameters) final;
+
+  std::shared_ptr<FunctionTree>
+  createFunctionTree(const ParameterList &DataSample, unsigned int pos,
+                     const std::string &suffix) const;
+
+protected:
+  /// Resonance spin
+  ComPWA::Spin J;
```

Comment:
This variable is not needed. Only use L.

Reply:
Yes, I will update it